tests/components/analytics/test_analytics.py

```diff
@@ -0,0 +1,245 @@
+"""The tests for the analytics ."""
+from unittest.mock import AsyncMock, Mock, PropertyMock, patch
+
+import aiohttp
+
+from homeassistant.components.analytics.analytics import Analytics
+from homeassistant.components.analytics.const import (
+    ANALYTICS_ENDPOINT_URL,
+    ATTR_PREFERENCES,
+    AnalyticsPreference,
+)
+from homeassistant.const import __version__ as HA_VERSION
+
+MOCK_HUUID = "abcdefg"
+
+
+async def test_no_send(hass, caplog, aioclient_mock):
+    """Test send when no prefrences are defined."""
+    aioclient_mock.post(ANALYTICS_ENDPOINT_URL, status=200)
+    analytics = Analytics(hass)
+    await analytics.load()
+    assert analytics.preferences == []
+
+    with patch("homeassistant.helpers.instance_id.async_get", return_value=MOCK_HUUID):
+        await analytics.send_analytics()
+    assert "Nothing to submit" in caplog.text
+    assert len(aioclient_mock.mock_calls) == 0
+
+
+async def test_load_with_supervisor_diagnostics(hass):
+    """Test loading with a supervisor that has diagnostics enabled."""
+    analytics = Analytics(hass)
+    assert AnalyticsPreference.DIAGNOSTICS not in analytics.preferences
+    with patch(
+        "homeassistant.components.hassio.get_supervisor_info",
+        side_effect=Mock(return_value={"diagnostics": True}),
+    ), patch(
+        "homeassistant.components.analytics.analytics.Analytics.supervisor",
```

Comment:
If we use `is_hassio` above we can patch that here instead to avoid patching details of the integration.